testnet/stacks-node/src/neon_node.rs

```diff
@@ -3595,7 +3794,12 @@ impl RelayerThread {
         if let Ok(miner_handle) = thread::Builder::new()
             .name(format!("miner-block-{}", self.local_peer.data_url))
             .stack_size(BLOCK_PROCESSOR_STACK_SIZE)
-            .spawn(move || miner_thread_state.run_tenure())
+            .spawn(move || {
+                if let Err(e) = miner_thread_state.send_mock_miner_messages() {
+                    warn!("Failed to send mock miner messages: {}", e);
+                }
```

Comment:
Should we run the tenure first, then do this mocking?

Reply:
I don't think it makes a difference either way. I can try moving it to after and see if things break (I think running the tenure does change internal state so not sure things will be instantiated still as they should be.)

Reply:
K I don't understand why, but putting it after breaks things as it fails to calculate the reward set in this case as expected. ... I am sure it is some other dtate issue that manifests down the line as this, but leaving as is for now to get this in asap.